druid/src/core.rs

```diff
@@ -30,6 +30,52 @@ use crate::{
 /// Our queue type
 pub(crate) type CommandQueue = VecDeque<(Target, Command)>;
 
+/// An object that can be used by a widget to obtain the focus and to handle keyboard events.
+#[derive(Debug, Clone, Copy)]
+pub struct FocusNode {
```

Comment:
I have very little information from these types as to what they represent or how they are used; we'll want to really figure out how to document this stuff before merging.

Reply:
I use this as an additional object to get information about the currently focused widget and about the current `FocusScope`.